test/functional/lcow_container_bench_test.go

```diff
@@ -0,0 +1,228 @@
+			id := cri_util.GenerateID()
+			scratch, _ := layers.ScratchSpace(ctx, b, vm, "", "", cache)
+			spec := oci.CreateLinuxSpec(ctx, b, id,
+				oci.DefaultLinuxSpecOpts(id,
+					ctrdoci.WithProcessArgs("/bin/sh", "-c", oci.TailNullArgs),
+					oci.WithWindowsLayerFolders(append(ls, scratch)))...)
+
+			c, _, cleanup := container.Create(ctx, b, vm, spec, id, hcsOwner)
+			if err := c.Start(ctx); err != nil {
+				b.Fatalf("could not start %q: %v", c.ID(), err)
+			}
+			init := cmd.Create(ctx, b, c, nil, nil)
+
+			b.StartTimer()
+			cmd.Start(ctx, b, init)
+			b.StopTimer()
+
+			cmd.Kill(ctx, b, init)
+			cmd.WaitExitCode(ctx, b, init, 137)
+
+			container.Kill(ctx, b, c)
+			container.Wait(ctx, b, c)
+			cleanup()
+		}
+	})
+
+	b.Run("InitExecKill", func(b *testing.B) {
+		vmID := strings.ReplaceAll(b.Name(), "/", "")
+		opts := defaultLCOWOptions(b, vmID)
+		cache := layers.CacheFile(ctx, b, "")
+		vm := uvm.CreateAndStartLCOWFromOpts(ctx, b, opts)
+		uvm.SetSecurityPolicy(ctx, b, vm, "")
+
+		b.StopTimer()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			id := cri_util.GenerateID()
+			scratch, _ := layers.ScratchSpace(ctx, b, vm, "", "", cache)
+			spec := oci.CreateLinuxSpec(ctx, b, id,
+				oci.DefaultLinuxSpecOpts(id,
+					ctrdoci.WithProcessArgs("/bin/sh", "-c", oci.TailNullArgs),
+					oci.WithWindowsLayerFolders(append(ls, scratch)))...)
+
+			c, _, cleanup := container.Create(ctx, b, vm, spec, id, hcsOwner)
+			init := container.Start(ctx, b, c, nil)
+
+			b.StartTimer()
+			cmd.Kill(ctx, b, init)
+			cmd.WaitExitCode(ctx, b, init, 137)
+			b.StopTimer()
+
+			container.Kill(ctx, b, c)
+			container.Wait(ctx, b, c)
+			cleanup()
+		}
+	})
+
+	b.Run("ContainerKill", func(b *testing.B) {
+		vmID := strings.ReplaceAll(b.Name(), "/", "")
+		opts := defaultLCOWOptions(b, vmID)
+		cache := layers.CacheFile(ctx, b, "")
+		vm := uvm.CreateAndStartLCOWFromOpts(ctx, b, opts)
+		uvm.SetSecurityPolicy(ctx, b, vm, "")
+
+		b.StopTimer()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			id := cri_util.GenerateID()
+			scratch, _ := layers.ScratchSpace(ctx, b, vm, "", "", cache)
+			spec := oci.CreateLinuxSpec(ctx, b, id,
+				oci.DefaultLinuxSpecOpts(id,
+					ctrdoci.WithProcessArgs("/bin/sh", "-c", "true"),
+					oci.WithWindowsLayerFolders(append(ls, scratch)))...)
+
+			c, _, cleanup := container.Create(ctx, b, vm, spec, id, hcsOwner)
+
+			// (c container).Wait() waits until the gc receives a notification message from
+			// the guest (via the bridge) that the container exited.
+			// The Linux guest starts a goroutine to send that notification (bridge_v2.go:createContainerV2)
+			// That goroutine, in turn, waits on the init process, which does not unblock until it has
+			// been waited on (usually via a WaitForProcess request) and had its exit code read
+			// (hcsv2/process.go:(*containerProcess).Wait).
+			//
+			// So ... to test container kill and wait times, we need to first start and wait on the init process
```

Comment:
So this just tests the time for the kill and wait command to return when the container init is already exited? I would think the thing we really care about is how long it takes to stop a simple program that's running.